feat(general): add stats to maintenance run

Lyndon-Li opened the pull request:
Maintenance is critical for healthy of the repository.
On the other hand, Maintenance is complex, because it runs multiple sub tasks each may generate different results according to the maintenance policy. The results may include deleting/combining/adding data/metadata to the repository.

It is worthy to add more observability for these tasks for below reasons:
1. It is helpful for troubleshooting. Any data change to the repository is critical, the observability info helps to understand what happened during the maintenance and why that happened
2. It is helpful for users to understand/predict the repo's behavior. The repo data may be stored in a public cloud for which costs are sensitive to scale/duration of data stored. On the other hand, repository has its own policy to manage the data, so the data is not deleted until it is safe enough according to the policy. The observability info helps users to understand how much data is in-use, how much data is out of use and when it is deleted

There will be a serial of PRs to add observability info for each sub task.

The current PR adds the interfaces, structures and workflows to save, retrieve and display the statistics info for maintenance tasks.

julio-lopez commented on repo/maintenancestats/types.go:
```diff
@@ -0,0 +1,28 @@
+}
+
+// Stats defines the methods for maintenance statistics.
+type Stats interface {
```
Either `Summarizer` or `ExtraSumarizer`
```suggestion
type Summarizer interface {
```

julio-lopez commented:
As soon as feedback is addressed, CI can run and this PR can be merged afterwards

- Ref: #4848

Lyndon-Li commented:
@julio-lopez 
I have addressed all the comments.

However, for "Add marshaling unit tests", I cannot add UT for the current PR.
The testing code are more like end to end function test instead of UT. So for example, for `TestMaintenanceInfoSimple`:

- I cannot control the the retrieved maintenance info, so cannot make sure the test covers every path of the changed code
- It is in a separate package `cli_test`, I cannot make a UT to test a private function `getMessageFromRun` in `cli` package

Which way do you prefer? And can you give some advice?

julio-lopez commented on repo/maintenancestats/types.go:
```diff
@@ -0,0 +1,42 @@
+func BuildExtra(stats Kind) (Extra, error) {
+	bytes, err := json.Marshal(stats)
+	if err != nil {
+		return Extra{}, errors.Wrapf(err, "error marshalling stats %v", stats)
```
```suggestion
		return Extra{}, errors.Wrapf(err, "error marshaling stats %v", stats)
```

Lyndon-Li commented:
> > However, for "Add marshaling unit tests", I cannot add UT for the current PR. The testing code are more like end to end function test instead of UT. So for example, for `TestMaintenanceInfoSimple`:
> > 
> > * I cannot control the the retrieved maintenance info, so cannot make sure the test covers every path of the changed code
> > * It is in a separate package `cli_test`, I cannot make a UT to test a private function `getMessageFromRun` in `cli` package
> > 
> > Which way do you prefer? And can you give some advice?
> 
> The suggestion is to have simple tests that ensure the struct marshaling/unmarshaling "round-trips", that is:
> 
> * instantiate the struct;
> * marshal it to JSON, it should not error out;
> * then unmarshal the resulting JSON back into a separate instance of the struct;
> * ensure that the values in the unmarshaled struct match the expected values.
> 
> These tests are simple, they **DO NOT** require a complex setup such to run maintenance, forcing it to create a schedule, etc.

Sure, so it is more like a UT. Will add it in the next PR.
